accelerator/hpu_accelerator.py

```diff
@@ -294,3 +294,22 @@ def build_extension(self):
 
     def export_envs(self):
         return []
+
+    def get_optimizer(self, optimizer_name, cpu_optimization, model_parameters, **optimizer_parameters):
+        from habana_frameworks.torch.hpex.optimizers import FusedAdamW
```

Comment:
I am not convinced that this much code is needed. 

It seems the goal is to replace AdamW on hpu accelerator as below:

`from deepspeed.ops.adam import FusedAdam` with 
`from habana_frameworks.torch.hpex.optimizers import FusedAdamW`

Is this correct?

Reply:
@tjruwase  Yes, but in addition to prepare the ground for other fused optimizers that are addressed in this function. for example: FusedLamb, OneBit, etc...
Do you have in mind something else?

Reply:
@nelyahu, thanks for the clarification. I do agree that deferring to accelerator to instantiate fused optimizer when available is the way to go. My concern is the duplication of optimizer selection logic here with `engine.py` will be difficult to maintain long term. 

I think the complication is that `engine.py` mixes optimizer name refinement (especially for Adam variants) and optimizer instantiation. By name refinement, I mean something like [ADAM_OPTIMIZER](https://github.com/microsoft/DeepSpeed/blob/aed599b4422b1cdf7397abb05a58c3726523a333/deepspeed/runtime/config.py#L72C1-L72C16) which is a user-facing config value, internally maps to one of many optimizers (e.g., `torch.optim.Adam`, `DeepSpeedCPUAdam`, etc.). So, my thought is to 
1. Create an internal naming convention for the various optimizer instantiations. For example, 

- `_TORCH_ADAM_OPTIMIZER` for `torch.optim.Adam`, 
- `_DS_FUSED_ADAM` for `deepspeed.ops.adam.FusedAdam`,
- `_DS_ADAM_OPTIMIZER` for `deepspeed.ops.adam.DeepSpeedCPUAdam`.  

2. Create a function that maps an external name to an internal one based on other configuration values.  For example, [ADAM_OPTIMIZER](https://github.com/microsoft/DeepSpeed/blob/aed599b4422b1cdf7397abb05a58c3726523a333/deepspeed/runtime/config.py#L72C1-L72C16) would map to one of `_TORCH_ADAM`, `_DS_FUSED_ADAM`, `_DS_CPUADAM`.

Based on the above the following changes could now be made:
1. Simplify `get_optimizer()` by accepting internal optimizer name and creating an optimizer if name matches its table of supported internal optimizer names. Also, `cpu_optimization` is no longer needed. 
2. Simplify [`_configure_basic_optimizer()`](https://github.com/microsoft/DeepSpeed/blob/aed599b4422b1cdf7397abb05a58c3726523a333/deepspeed/runtime/engine.py#L1276) to (i) convert to internal name, (ii) call `accelerator().get_optimizer()` with internal name to create optimizer, and (iii) otherwise fall through to existing optimizer creation logic. 

Looking forward to your feedback. Thanks!

Reply:
@tjruwase I applied the changes you suggested, however some optimizers still require specific code. Which one of the commits you think is best?

Reply:
@ShellyNR, thanks for making the changes. This looks better to me. 

When you say 'some optimizers still require specific code`, is that referring to `_ADAM` in `hpu_accelerator` which maps to `torch.optim.Adam()`? If so, I think that is fine. My goal is to avoid duplication between accelerator codes and `engine.py`.